packages/core/content-manager/server/src/history/services/history.ts

```diff
@@ -107,8 +107,7 @@ const createHistoryService = ({ strapi }: { strapi: LoadedStrapi }) => {
     async findVersionsPage(params: HistoryVersions.GetHistoryVersions.Request['query']) {
       const [{ results, pagination }, localeDictionary] = await Promise.all([
         query.findPage({
-          page: 1,
-          pageSize: 10,
+          ...params,
```

Comment:
are the defaults set on the query layer?

On the content manager side normally these values are send to a yup validation, that enforces minimum and maximum values (for example trying to request 1000000 elements).

If it's possible I think it might be useful to yup validate the params :D

Reply:
Yes regarding the defaults I found [this](https://github.com/strapi/strapi/blob/614d53bc166daa494493a7cb49a2cc333f754a80/packages/core/database/src/entity-manager/entity-repository.ts#L6) for the entity service, so I assume the same is done for the query layer. And requests from the frontend will always be paginated so I don't think we need to care about what the default is (at long as there's one)

Good point about Yup for min and max

And would you agree that these validations should happen in the controller, and that the service can trust what we give it? Or do you think it should be in the service?

Reply:
good points, and looks like the db does the same :) https://github.com/strapi/strapi/blob/eb64ae5745bd7585cc21514afbf2d49f39961d94/packages/core/database/src/entity-manager/entity-repository.ts

>And would you agree that these validations should happen in the controller, and that the service can trust what we give it? Or do you think it should be in the service?
I believe those should be on the controller layer yep! as each interface might expose different constraints, and the bussiness logic should expect sanitized and validated data